### PR TITLE
Add minor WSL building updates

### PR DESCRIPTION
**Pull request by chromatic**
Adding notes here; we should get feedback from other developers if this works for them as well.

### CODE REVIEW

**Comment by UsaRandom** [comment]
You can determine WSL version by running in windows cmd prompt:

```
wsl -l -v
```

**Comment by MishaBoar** [comment]
I tested the instructions, without issues. Tested also on a WSL install running 20.04.6 LTS. 

> You can determine WSL version by running in windows cmd prompt:
> 
> ```
> wsl -l -v
> ```

Yeah, this is useful.

I asked my colleague, who is a graphic designer with limited development experience, to try the guide and he also had no issues. Only doubts he had were about the WSL version and the selection of the posix option (after a few moments, he figured it out).

File: doc/build-windows.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ To get the bash shell, you must first activate the feature in Windows.
 2. Install Ubuntu
   * Open Microsoft Store and search for "Ubuntu 18.04" or use [this link](https://www.microsoft.com/store/productId/9N9TNGVNDL3Q)
```
Comment (patricklodder) [inline doc/build-windows.md:34]: May want to update this and on line 38, to focal/20.04, as the depends system works on that and Bionic is EOL.

**Comment by patricklodder** [comment]
> needed to add sudo to [..]

this is because you created the source directory with `sudo`. Instead of working from `/usr/src`, you may just work from `$HOME` and do the git clone there. This should get rid of all the `sudo` during compilation

**Review by edtubbs — CHANGES_REQUESTED** [review]
Reviewed updates and built both 64-bit and 32-bit versions; see results below. Recommend adding an `update-alternatives` step to the 32-bit build.

## w/intel-ipsec-mb compiled on wsl 1 bionic (64-bit)

```
ed@DESKTOP-OPJD6CK:~/dogecoin/depends$ make HOST=x86_64-w64-mingw32 AVX2=1 -j 4
...
Postprocessing intel-ipsec-mb...
Caching intel-ipsec-mb...
copying packages: native_ccache native_protobuf native_nasm boost openssl libevent zeromq intel-ipsec-mb qrencode protobuf zlib qt bdb miniupnpc
to: /home/ed/dogecoin/depends/x86_64-w64-mingw32
ed@DESKTOP-OPJD6CK:~/dogecoin/depends$ cd ..
ed@DESKTOP-OPJD6CK:~/dogecoin$ CONFIG_SITE=$PWD/depends/x86_64-w64-mingw32/share/config.site ./configure --prefix=/ --enable-experimental --enable-scrypt-sse2 --with-intel-avx2 --with-gui=qt5 --enable-reduce-exports
...
Options used to compile and link:
  with wallet   = yes
  with gui / qt = yes
    qt version  = 5
    with qr     = yes
  with zmq      = yes
  with test     = yes
  with bench    = yes
  with upnp     = yes
  debug enabled = no
  werror        = no

  experimental  = yes
    SSE2 Scrypt   = yes
    AVX2 crypto   = yes
    ARMv8 crypto  = no
    ARMv82 crypto = no

  target os     = windows
  build os      = 

  CC            = /home/ed/dogecoin/depends/x86_64-w64-mingw32/share/../native/bin/ccache x86_64-w64-mingw32-gcc
  CFLAGS        = -pipe -O2 
  CPPFLAGS      = -I/home/ed/dogecoin/depends/x86_64-w64-mingw32/share/../include/  -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS -D_MT -DWIN32 -D_WINDOWS -DBOOST_THREAD_USE_LIB -D_FILE_OFFSET_BITS=64
  CXX           = /home/ed/dogecoin/depends/x86_64-w64-mingw32/share/../native/bin/ccache x86_64-w64-mingw32-g++ -std=c++11
  CXXFLAGS      = -pipe -O2  -fvisibility=hidden
  LDFLAGS       = -L/home/ed/dogecoin/depends/x86_64-w64-mingw32/share/../lib 

ed@DESKTOP-OPJD6CK:~/dogecoin$ make -j 4
...
  CXXLD    test/test_dogecoin.exe
  CXXLD    bench/bench_dogecoin.exe
  AR       qt/libdogecoinqt.a
/usr/bin/x86_64-w64-mingw32-ar: `u' modifier ignored since `D' is the default (see `U')
  OBJCXXLD qt/dogecoin-qt.exe
  CXXLD    qt/test/test_dogecoin-qt.exe
make[2]: Leaving directory '/home/ed/dogecoin/src'
make[1]: Leaving directory '/home/ed/dogecoin/src'
Making all in doc/man
make[1]: Entering directory '/home/ed/dogecoin/doc/man'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/home/ed/dogecoin/doc/man'
make[1]: Entering directory '/home/ed/dogecoin'
make[1]: Nothing to be done for 'all-am'.
make[1]: Leaving directory '/home/ed/dogecoin'
ed@DESKTOP-OPJD6CK:~/dogecoin$ time src/test/test_dogecoin.exe 
Running 254 test cases...

*** No errors detected

real    1m13.347s
user    0m0.000s
sys     0m0.000s
ed@DESKTOP-OPJD6CK:~/dogecoin$ src/bench/bench_dogecoin.exe 
#Benchmark,count,min,max,average,min_cycles,max_cycles,average_cycles
Base58CheckEncode,147456,0.000005703244824,0.000010734423995,0.000007121765520,13695,21206,15063
Base58Decode,589824,0.000001447831892,0.000002353775926,0.000001758843508,3275,4984,3748
Base58Encode,589824,0.000001578369847,0.000002790417057,0.000001839658782,3527,5327,3891
CCheckQueueSpeed,8192,0.000104404520243,0.000207992270589,0.000127987674205,245296,446633,270268
CCheckQueueSpeedPrevectorJob,352,0.002656497061253,0.003552496433258,0.002895715561780,5614075,6524727,6057034
CCoinsCaching,180224,0.000005284789950,0.000007167356671,0.000005987082576,11891,14407,12533
CoinSelection,576,0.001394763588905,0.006425499916077,0.001831687572930,3121512,12449428,3881712
LockedPool,576,0.001631390303373,0.002625197172165,0.001928222262197,3701288,5488838,4070400
MempoolEviction,22528,0.000033777207136,0.000101812183857,0.000046572446908,77545,191131,98607
RIPEMD160,480,0.001732781529427,0.002401471138000,0.002110194166501,4161563,5203329,4501247
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.001064000000000,0.001064000000000,0.001064000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.001069000000000,0.001069000000000,0.001069000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom-refresh,1,0.000000000000000,0.000000000000000,0.000000000000000
RollingBloom,655360,0.000001217893441,0.000002103333827,0.000001569525193,2560,4432,3314
SHA1,640,0.001409750431776,0.002051010727882,0.001696082949638,2981965,4379773,3585424
SHA256,352,0.002604156732559,0.003715060651302,0.002994298257611,5479649,7248905,6297033
SHA256_32b,6,0.190112590789795,0.208786487579346,0.199377179145813,411139738,434255959,421663634
SHA512,576,0.001440808176994,0.002994000911713,0.001894440915849,3575123,6302708,4009455
Scrypt,5632,0.000154425855726,0.000258663669229,0.000183952972293,370995,547234,392586
SipHash_32b,44,0.020419955253601,0.027857482433319,0.024022497914054,44734713,56804811,50658478
Sleep100ms,10,0.105425477027893,0.111533045768738,0.107882213592529,220265894,235636481,226916881
Trig,25165824,0.000000033342531,0.000000050283461,0.000000040111303,76,110,84
ed@DESKTOP-OPJD6CK:~/dogecoin$ 
```


## w/o experimental compiled on wsl 1 bionic (32-bit)

```
ed@DESKTOP-OPJD6CK:~/dogecoin/depends$ make HOST=i686-w64-mingw32 -j 4
...
copying packages: native_ccache native_protobuf boost openssl libevent zeromq qrencode protobuf zlib qt bdb miniupnpc
to: /home/ed/dogecoin/depends/i686-w64-mingw32
ed@DESKTOP-OPJD6CK:~/dogecoin/depends$ cd ..
ed@DESKTOP-OPJD6CK:~/dogecoin$ CONFIG_SITE=$PWD/depends/i686-w64-mingw32/share/config.site ./configure --prefix=/ --with-gui=qt5 --enable-reduce-exports
Options used to compile and link:
  with wallet   = yes
  with gui / qt = yes
    qt version  = 5
    with qr     = yes
  with zmq      = yes
  with test     = yes
  with bench    = yes
  with upnp     = yes
  debug enabled = no
  werror        = no

  experimental  = no

  target os     = windows
  build os      = 

  CC            = /home/ed/dogecoin/depends/i686-w64-mingw32/share/../native/bin/ccache i686-w64-mingw32-gcc
  CFLAGS        = -pipe -O2 
  CPPFLAGS      = -I/home/ed/dogecoin/depends/i686-w64-mingw32/share/../include/  -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS -D_MT -DWIN32 -D_WINDOWS -DBOOST_THREAD_USE_LIB -D_FILE_OFFSET_BITS=64
  CXX           = /home/ed/dogecoin/depends/i686-w64-mingw32/share/../native/bin/ccache i686-w64-mingw32-g++ -std=c++11
  CXXFLAGS      = -pipe -O2  -fvisibility=hidden
  LDFLAGS       = -L/home/ed/dogecoin/depends/i686-w64-mingw32/share/../lib  -Wl,--large-address-aware

ed@DESKTOP-OPJD6CK:~/dogecoin$ make -j 4
...
  CXXLD    test/test_dogecoin.exe
  CXXLD    bench/bench_dogecoin.exe
  AR       qt/libdogecoinqt.a
/usr/bin/i686-w64-mingw32-ar: `u' modifier ignored since `D' is the default (see `U')
  OBJCXXLD qt/dogecoin-qt.exe
  CXXLD    qt/test/test_dogecoin-qt.exe
make[2]: Leaving directory '/home/ed/dogecoin/src'
make[1]: Leaving directory '/home/ed/dogecoin/src'
Making all in doc/man
make[1]: Entering directory '/home/ed/dogecoin/doc/man'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/home/ed/dogecoin/doc/man'
make[1]: Entering directory '/home/ed/dogecoin'
make[1]: Nothing to be done for 'all-am'.
make[1]: Leaving directory '/home/ed/dogecoin'
ed@DESKTOP-OPJD6CK:~/dogecoin$ 
```

### 32-bit test_dogecoin.exe on Windows 7
test

### 32-bit bench_dogecoin.exe on Windows 7
bench

### 32-bit dogecoin-qt.exe on Windows 7
qt

**Comment by chromatic** [comment]
> git clone https:/ /githu b.com/dogecoin/dogecoin.git ( delete spaces )
> git checkout 1.14.7-dev

I added a note about git checkouts here and recommend against the use of `sudo`. Thanks for the suggestion!

File: doc/build-windows.md [hunk 2]
```diff
@@ -85,25 +103,31 @@ If using WSL 1 then build using:
...
+report any errors with this in mind.
+
+Also be careful using the `-j` flag with any `make` command. People have reported strange compilation errors when doing so.
+
 ## Building for 32-bit Windows
 
 To build executables for Windows 32-bit, install the following dependencies:
 
-    sudo apt-get install g++-mingw-w64-i686 mingw-w64-i686-dev 
+    sudo apt-get install g++-mingw-w64-i686 mingw-w64-i686-dev
```
[the remark this answers is not in the document]
Comment (chromatic) [reply doc/build-windows.md:130]: Good idea. Thank you!

**Comment by FierceSkit** [comment]
Bildschirmfoto 2024-02-26 um 00 58 10

**Review by patricklodder — APPROVED** [review]
ACK - code reviewed.

Tested by:

- @Schmeckl3s
- @nowon
- @MishaBoar
- @fultondoge 
- @edtubbs 
- @FierceSkit 

Thanks everyone!

**Pull request merged**